File: databasesCoursework/upsert_review.js
function insert_review(id, mark, text, date, author, film_id) {
    var film = db.films.findOne(
        { "_id": film_id } );

    if(typeof film == 'undefined' || film == null) {
        print("Invalid film id.");
        return;
    }

    db.reviews.update(
        {_id: id},
        {
            $set: {
                _id: id,
                mark: mark,
                text: text,
                date: date,
                author: author,
                film_id: film_id
            }
        }, {upsert: true});

    db.films.update(
        { _id: film_id },
        { $addToSet: { reviews: id } }
    )
}

File: databasesCoursework/upsert_review.lua
local Count = 0
						--film id
for Index, Value in pairs(redis.call('hgetall', 'film:' .. KEYS[5])) do
  Count = Count + 1
end
if Count == 0 then 
	return -1
end

Count = 0
						--review id
for Index, Value in pairs(redis.call('hgetall', 'review:' .. KEYS[6])) do
  Count = Count + 1
end
if Count == 0 then
	Count = redis.call('get', 'counters:reviews')
else Count = KEYS[6]
end

redis.call('hmset', 'review:' .. tostring(Count), 'author', KEYS[1],
'mark', KEYS[2], 'date', KEYS[3], 'text', KEYS[4], 'film_id', KEYS[5]);

redis.call('set', 'counters:reviews', Count + 1);

return Count

